### ISSUE TITLE
Impossible to acces at Odoo V8 with Google Chrome 50

### ISSUE THREAD

**Issue by cedric190985** [odoo/odoo#11629]
Impacted versions:
Odoo V8

Steps to reproduce:
1 Use Chrome 50 Beta 
2 Go to http://runbot.odoo.com/runbot
3 Click to 8.0 on the icon blue

Current behavior:
Blank Page

Expected behavior:
A data is display

Video/Screenshot link (optional):

**Comment by bradhaywood** [comment]
Running Chrome 50 on ChromeOS and having a similar problem. Once logged in, I can see the menu at the top, but the page content is actually blank. Only started happening once I upgraded from 49 to 50.

**Comment by phoenixsampras** [comment]
Same problem with Chrome 50.

Works fine with Chrome 49.

**Comment by phoenixsampras** [comment]
Same problem with Odoo 9

**Comment by Julien-BARRAU** [comment]
Update automatic today on Chrome : Version 50.0.2661.75 m

Odoo v8 is not working anymore ! 

We have installed Firefox for now

**Comment by qndi** [comment]
Some problem here.
It is strange that there is not more information and communication about this issue, since it seams to effect every Odoo User on V8 who is running v8 with the newest Google Chrome Update.

**Comment by takis** [comment]
Same problem here. I'd reported it to Google months ago (as I also use both Chrome Canary and daily builds) using the in-browser "Report an issue..." menu, but never got any reply.

**Comment by phoenixsampras** [comment]
Same Problem,

Odoo 8 and 9 is useless in Chrome 50+

Fix the issue ODOO SA!!

**Comment by qndi** [comment]
I tried you apply the patch on saas-6 without success. Do you also have a patch for that version?

**Comment by sle-odoo** [comment]
The issue doesn't appear on the last saas-6 on runbot. In this version, the scrolling parts of the webclient are using flex and not table layouts. Are you sure it's related?

**Comment by agcopley** [comment]
The fixes above do not address the issue in the POS. A blank screen is still showing

**Comment by phoenixsampras** [comment]
HELP!!

Odoo8 and Odoo9 pos doenst work anymore!!

http://prnt.sc/as9q4t

Emergency!!

**Comment by spadae22** [comment]
We are seeing many issue on odoo v8. I updated the patch but we see things not lining up.
![screen shot 2016-04-14 at 3 25 53 pm](https://cloud.githubusercontent.com/assets/7741100/14540866/3e00a1ae-0255-11e6-8a39-ec01f6d6a1e2.png)

**Comment by lmignon** [comment]
@Yenthe666 This fix breaks IE 11.0.9600.18.282. Once applied, the application no more works on IE. Only the main menu is displayed.

**Comment by Yenthe666** [comment]
@lmignon thanks for the heads-up. Looks like Odoo has quite some CSS issues to fix now!

**Comment by sle-odoo** [comment]
@spadae22 this is not related. let's kee focus on the scrollable div not so scrollable according to different browsers..

@lmignon arg you are right. i tested the patched only on ie9 where it is working as expected. note that the layout is broken differently in ie10..

@agcopley i confirm that the point of sale is also broken. looks lined to table-row, table-cell and distribution of 100% height too.

**Comment by sle-odoo** [comment]
There's an [issue in the Chromium bug tracker](https://bugs.chromium.org/p/chromium/issues/detail?id=603507&q=component%3ABlink%3ELayout%3ETable&colspec=ID%20Pri%20M%20Stars%20ReleaseBlock%20Component%20Status%20Owner%20Summary%20OS%20Modified).

**Comment by wtaferner** [comment]
@sle-odoo 
No official confirmation from any Google employee yet, right? But it really seems that Chrome did a major fail on the update...unless they did not implement any official standard which was ignored by all browsers until now and they want to force the whole net to adapt ;-)

**Comment by gerencia** [comment]
Same here, problems with Canary wit Odoo8 portal too, after only update Chrome to versión

**Comment by aek** [comment]
I solve it using this quick fix, hope that helps to find the complete solution. I add this css rules in one of my modules and make it works mostly everything. Just the database manager view menu continue hide because that view works without a database so no modules are loaded and for that it works by adding the css rules on the base.css of the web module(I tried using a module with a server_wide_modules or --load options but no luck).
The css rules are:

``` css
.openerp .oe_leftbar > div > div > div {
    min-height: 450px;
}

.openerp .oe_application > div > .oe_view_manager > .oe_view_manager_wrapper > div{
    min-height: 500px;
}

.openerp .oe_leftbar > div > div > div {
    min-height: 323px;
}
```
This it's tested on Odoo 8, the same should apply to Odoo 9 or others versions. This solution of course could be improved for anyone with more deep knowledge of css than me but it works. I hope this guide to the final solution

**Comment by hardikns** [comment]
I think this would be a cleaner solution. Not sure if it breaks in IE or Safari. 

```

.openerp .oe_application > div > .oe_view_manager > .oe_view_manager_wrapper > div{
   display:table-cell;
}

.openerp .oe_leftbar > div > div > div {
    display:table-cell;
}
```

Can something similar be included in the base css of the web?

**Comment by aek** [comment]
That works too!!

**Comment by sle-odoo** [comment]
@hardikns please look at the patch that was reverted. this works fine in chrome 50 but breaks internet explore 9, 10 and 11.

@aek not tested but: you set a min height because the height is not properly passed from the parents, so indeed you get a scrollable zone but not from below the bottom searchview to the bottom of the screen.

@wtaferner while looking at the but reports i saw [this one](https://bugs.chromium.org/p/chromium/issues/detail?id=583670&q=label%3ACr-Blink-Layout-Table&colspec=ID%20Pri%20M%20Stars%20ReleaseBlock%20Cr%20Status%20Owner%20Summary%20OS%20Modified) and the author of the patch said: "It got us closer to the css spec". Reference of [this commit](https://chromium.googlesource.com/chromium/src/+/8876584335b48c99cf8df552ef4d8efebb131041), The change does not directly apply to our breakage but maybe it's a side effect?

**Comment by wtaferner** [comment]
@sle-odoo 
yep, I saw this one too and had the same conclusion as you had...it is a really bad situation and I would suggest to implement a workaround for Chrome (remembers me a little bit on IE6-8 times whereas I have more trust in Google) which can be reverted later when Google Devs might consider and "repair". 
In case they insist on their view about the specs which might be correct, Odoo can improve their implementation anyway. 
It is anyway a blocking issue which needs to be tackled real quick...I informed my customers not to update to version 50 and use Firefox as a fallback but this is something users normally do not understand and tend to have irrational conclusions about browsers and/or the software in use.

**Comment by aek** [comment]
@sle-odoo Indeed I don't get into the bottom of the screen but could be the start for a way to get it calculated using something like resize of the screen and onload events to calculate the height to the bottom of the screen. I just seek for a quick way to get it work because my customers that have been updated to Chrome 50 see nothing when open their Odoo instance

**Comment by thomich** [comment]
for customers that use chrome, use this fix in a own module: https://github.com/odoo/odoo/commit/00ead29c7c1877de94be8c28bd587fa7acedcc69

if ie is used at customer, use firefox, chromium or vivaldi...

**Comment by phoenixsampras** [comment]
What about POS?? pos in Odoo8 and Odoo9 still broken

**Comment by aek** [comment]
I test this css rules using my approach with pos and works:
```css
.pos .subwindow .subwindow-container-fix {
    min-height: 186px;
}
.pos .subwindow .subwindow-container-fix.screens{
    min-height: 608px;
}
.pos .screen .content-cell .content-container {
    min-height: 559px;
}
```
To load those in a custom module you will need something like this:
```xml
<?xml version="1.0" encoding="utf-8"?>
<openerp>
    <data>
        <template id="assets_backend" name="solt_web assets" inherit_id="point_of_sale.index">
            <xpath expr="//head" position="inside">
                <link rel="stylesheet" href="/solt_pos/static/src/css/styles.css"/>
            </xpath>
        </template>
    </data>
</openerp>
```
Otherwise it will not load your css

**Comment by phoenixsampras** [comment]
Works in POS, Odoo9 and Odoo8

Thank bro!

**Comment by Yenthe666** [comment]
@sle-odoo any status update on this? In Odoo 8 there are a LOT of issues with Odoo V8.
The POS is rendered useless and almost all views are completely broken. Example:
![image](https://cloud.githubusercontent.com/assets/6352350/14599719/43d4704a-0559-11e6-99f8-e31cdeca8c49.png)

**Comment by odony** [comment]
@Yenthe666 yes we're actively working on it. In the mean time please have customers using an older version of Chrome, or temporarily switch to Firefox/IE or applying patch 00ead29c7c1877de94be8c28bd587fa7acedcc69 (if no IE users).
Patches for POS and web client are both upcoming, with compatibility shims for IE.

**Comment by odony** [comment]
The issue should now be fixed in web client 8.0 with patch f992c8e.
As for the POS, it is actually broken in Chrome 50 in all versions: 7.0, 8.0, 9.0! Different fixes are being tested at the moment by @jorenvo.

Note: it doesn't seem the Chrome developers realized the complete side-effects of their change. They apparently tried to mimic Firefox, but Firefox handles [the bug they were trying to fix](https://bugs.chromium.org/p/chromium/issues/detail?id=353580) *without* breaking this [classical layout](http://jsfiddle.net/257zsmaz/), used by Odoo and many other web apps.

**Comment by odony** [comment]
note: [upstream issue](https://bugs.chromium.org/p/chromium/issues/detail?id=604173#c4) for tracking the regression apparently introduced by [this commit](https://chromium.googlesource.com/chromium/src/+/8876584335b48c99cf8df552ef4d8efebb131041)

**Comment by jorenvo** [comment]
v7 POS fixed at 09a9e4aeb57b92b03ba73f831d8943ec24cb9589. Fixes for other versions are slightly more complicated and will follow shortly.

**Comment by mariuszmizgier** [comment]
Not sure that whether it was already spotted, but now new message window is broken - screenshot attached.

![image](https://cloud.githubusercontent.com/assets/7665343/14617852/31902490-05b0-11e6-96df-b9d84f99b80f.png)

**Comment by phoenixsampras** [comment]
same here!! help!!

**Comment by jorenvo** [comment]
v8 POS fixed at 9fa53b7ee7fc8faca2ee8bc1d5205dca62f3e565. Fixes for later versions will follow shortly.

**Comment by jorenvo** [comment]
saas-6 POS fixed at 77dae4c9c54ac8341867f7b382a9be564f23d716. Fixes for later versions will follow shortly.

**Comment by jorenvo** [comment]
v9 POS fixed at 4ff34a98bcefa96c908baae1646a5af8acd21913. This patch will forward-port cleanly to all later versions of Odoo.

**Comment by thomich** [comment]
there are more layout issues with this chrome 50 update.. and i think there are a lot of odoo layout issues following.. here one of it:

sale
![selection_002](https://cloud.githubusercontent.com/assets/4398840/14638393/80835db0-0637-11e6-977e-eed37a1633c1.png)

purchase
![selection_003](https://cloud.githubusercontent.com/assets/4398840/14638399/85d2b69e-0637-11e6-8c17-eec50484047b.png)

there will be more views... eg. more in purchase, warehouse,...

so, it's not really fixed for us...

**Comment by odony** [comment]
Hopefully the latest version of the fix at 7f6bd2dc666ca93c09df3131da2866ae2e4b13fa should solve the issue in chrome 50 without the side-effects of breaking modal windows (e.g. mail composer) or x2m view managers, as reported by @thomich and @mariuszmizgier.
All issues (including all POS versions) seem to be fixed now, so I'm closing the issue again.
Thanks for your feedback and patience!

BTW the upstream issue is still open [here](https://bugs.chromium.org/p/chromium/issues/detail?id=603507), so there is still hope to see it fixed in Chrome.

**Comment by spadae22** [comment]
@odony Yes Chrome should fix the bug as I read your report as well as others.

**Comment by carllhw** [comment]
Database Manager 
http://143105-8-0-staging-odo-7f6bd2.runbot8.odoo.com/web/database/manager#action=database_manager
![image](https://cloud.githubusercontent.com/assets/9696301/14664790/92071bde-0700-11e6-9d9c-7548e2af98ae.png)

**Comment by jorenvo** [comment]
Seems like this particular change is being a considered a bug by Google and is being reverted [1], which is good news.

[1] https://bugs.chromium.org/p/chromium/issues/detail?id=603507#c24

**Comment by mcarropellicer** [comment]
So what should we do? Patch our instance or wait for the Chrome update?

TBH i've patched our instance and everything seems to work fine, should i revert the patches once Chrome is patched?

Thanks

**Comment by jorenvo** [comment]
We plan on waiting until the revert makes it into the stable Chrome release on all platforms. After that we will wait a couple of weeks so everyone's updated and revert our patches that workaround the bug. The POS patches don't break anything but I'd like to get rid of them as they're ugly and they won't have any impact anymore after Chrome is fixed.

**Comment by mcarropellicer** [comment]
Thank you so much @jorenvo

**Comment by odony** [comment]
@carllhw thanks, indeed the fix needs to be applied to the database manager screen as well, will fix it shortly.

So the problematic commit was [just reverted](https://chromium.googlesource.com/chromium/src/+/d5834ce2cf3bbaf98f8eff1a8ed0e0df0f085bc4) upstream, and is included as of Chrome build 50.0.2661**.86**.
This release is not blessed as stable yet (on most platform it's  50.0.2661.75), so we'll have to wait until it is by watching the [Chrome release dashboard](http://omahaproxy.appspot.com/). Then as @jorenvo says, we'll wait a bit and revert all relevant commits, as they are strictly unnecessary.

Like @mcarropellicer, if you've got the problem now you should update to the latest Odoo 8 version (or apply the patches), and when you next update Chrome will be fixed and the fixes will be gone from Odoo to.

**Comment by odony** [comment]
@carllhw The db manager issue should be fixed by fc2aa7360691a7b6075337a4ee9cda762a98b433.

**Comment by fernandomr** [comment]
The fix sugested by @odony works!
Just a note, the datepicker is a little bit unformatted. It seems another thing that the Chrome update caused.
![captura de tela de 2016-04-20 15 47 30](https://cloud.githubusercontent.com/assets/4227936/14686526/78c483c8-070f-11e6-8763-f16081cf3021.png)

**Comment by spadae22** [comment]
Is Chrome going to fix this mess they created for odoo any other site? I look at the bug report and it seems like "well" whatever

**Comment by thomich** [comment]
btw... chrome was updated this night:
google-chrome-stable:
  Installed: 50.0.2661.86-1
  Candidate: 50.0.2661.86-1
  Version table:
 *** 50.0.2661.86-1 0
        500 http://dl.google.com/linux/chrome/deb/ stable/main amd64 Packages
        100 /var/lib/dpkg/status

**Comment by odony** [comment]
Status update: according to their [dashboard](http://omahaproxy.appspot.com/), the latest Chrome 50 update on Linux and Mac is 50.0.2661.86 and has the fix already. All Windows versions are still broken (2661.75), and the IOS build is now broken (used to be Chrome 49 until today: 50.0.2661.77).

**Comment by odony** [comment]
@fernandomr The graphical glitch in the datepicker was visible before Chrome 50, so it seems to be a different issue.

**Comment by spadae22** [comment]
@odony I see Chrome we updated with the bug reverted. So are you reverting the revisions?

**Comment by odony** [comment]
@spadae22 Not yet, as explained in previous comments.

**Comment by alyngsoe** [comment]
This works for me (Windows 8.1):  (I had the same problems as described)
1. removed old chrome  (newest 32bit version)
2. restart
3. installed new 64bit version   https://www.google.com/chrome/browser/desktop/index.html

**Comment by Yenthe666** [comment]
@alyngsoe it is enough to go to settings > about and then Chrome will automatically update.
A simple reloading of the browser and you have the latest version, which indeed fixes everything again.

**Comment by odony** [comment]
Both 32bit and 64bit versions of Chrome on Windows have the fixed build 50 (50.0.2661.87) as of last night, so anyone affected by the bug should simply update to the latest version.
We'll wait until next week so the majority of users who ended up with the broken Chrome 50 builds will have auto-updated to a fixed one, and then we'll revert the bugfixes.

**Comment by mcarropellicer** [comment]
As additional info, i've reverted the patches in our instance and everything works fine so far, thanks for addressing the problem.

Chrome Version 50.0.2661.87 m
